Documentation/user-guides/configuring-cluster-monitoring.md

```diff
@@ -1,137 +1,858 @@
+      tolerations:
+        - operator: "Exists"
+      volumeClaimTemplate:
+        spec:
+          resources:
+            requests:
+              storage: 15Gb
 ```
-### NodeExporterConfig
 
-Use NodeExporterConfig to configure parameters for deployment of the `node-exporter` components.
+## Common data types
+
+This section contains a set of data types which common to both configurations.
+
+### AdditionalAlertmanagerConfig
+
+Holds configuration about how component should communicate with aditional Alertmanager instances, from configuring the URL scheme, timeout to authentication method and others.
+
+```yaml
+additionalAlertmanagerConfigs:
+  # The api version of Alertmanager.
+  apiVersion: <string>
+  # Bearer token to use when authenticating to Alertmanager.
+  bearerToken: v1.SecretKeySelector
+  # Path prefix to add in front of the push endpoint path.
+  pathPrefix: <string>
+  # The URL scheme to use when talking to Alertmanagers.
+  scheme: <string>
+  # List of statically configured Alertmanagers.
+  staticConfigs: <[]string>
+  # The timeout used when sending alerts.
+  timeout: <string>
+  # TLS Config to use for alertmanager connection.
+  tlsConfig: TLSConfig
+```
+
+#### TLSConfig
+
+Holds configuration for TLS configuration
+
+```yaml
+tlsConfig:
+  # The CA cert in the Prometheus container to use for the targets.
+  ca: v1.SecretKeySelector
+  # The client cert in the Prometheus container to use for the targets.
+  cert: v1.SecretKeySelector
+  # Disable target certificate validation.
+  insecureSkipVerify: <bool>
+  # The client key in the Prometheus container to use for the targets.
+  key: v1.SecretKeySelector
+  # Used to verify the hostname for the targets.
+  serverName: <string>
+```
+
+### RemoteWriteSpec
+
+Holds configuration for remote write
+
+```yaml
+remoteWrite:
+  # The URL of the endpoint to send samples to.
+  url: <string>
+  # The name of the remote write queue, must be unique if specified. The
+  # name is used in metrics and logging in order to differentiate queues.
+  # Only valid in Prometheus versions 2.15.0 and newer.
+  name: <string>
+  # Timeout for requests to the remote write endpoint.
+  remoteTimeout: <string>
+  # Custom HTTP headers to be sent along with each remote write request.
+  # Be aware that headers that are set by Prometheus itself can't be overwritten.
+  # Only valid in Prometheus versions 2.25.0 and newer.
+  headers: <map[string]string>
+  # The list of remote write relabel configurations.
+  writeRelabelConfigs: RelabelConfig
+  # BasicAuth for the URL.
+  basicAuth: BasicAuth
+  # Bearer token for remote write.
+  bearerTokenFile: <string>
+  # TLS Config to use for remote write.
+  tlsConfig: SafeTLSConfig
+  # Optional ProxyURL
+  proxyUrl: <string>
+  # QueueConfig allows tuning of the remote write queue parameters.
+  queueConfig: QueueConfig
+  # MetadataConfig configures the sending of series metadata to remote storage.
+  metadataConfig: MetadataConfig
```

Comment:
from https://github.com/openshift/cluster-monitoring-operator/blob/release-4.11/pkg/manifests/config.go#L155-L188
there are also `authorization`, `sigv4`, `oauth2`

Reply:
Yes, I didn't include them since this was mostly to fulfil user documentation for 4.10. Once @bburt-rh is finished I'll add those missing fields